rosidl_cmake/cmake/rosidl_generate_interfaces.cmake

```diff
@@ -283,7 +283,26 @@ macro(rosidl_generate_interfaces target)
     list(APPEND rosidl_generate_interfaces_ABS_IDL_FILES "${_abs_idl_file}")
   endforeach()
 
-  ament_execute_extensions("rosidl_generate_idl_interfaces")
+  # clean extension points include guards
+  set(_extension_point "rosidl_generate_idl_interfaces")
+  if(AMENT_EXTENSIONS_${_extension_point})
+    foreach(_extension ${AMENT_EXTENSIONS_${_extension_point}})
+      string(REPLACE ":" ";" _extension_list "${_extension}")
+      list(LENGTH _extension_list _length)
+      if(NOT _length EQUAL 2)
+        message(FATAL_ERROR "ament_execute_extensions(${_extension_point}) "
+          "registered extension '${_extension}' can not be split into package "
+          "name and cmake filename")
+      endif()
+      list(GET _extension_list 1 _cmake_filename)
+      string(TOUPPER "${_cmake_filename}" _cmake_filename_upper)
+      get_filename_component(_generator_guard "${_cmake_filename_upper}" NAME_WE)
+      unset(_${_generator_guard})
+    endforeach()
+  endif()
```

Comment:
This should not be necessary and is a pretty ugly hack imo.

Reply:
Done